### PR TITLE
docs(samples): add iam deny samples and test

### PR DESCRIPTION
**Pull request by Sita04**
Added samples and tests for IAM Deny launch.

### CODE REVIEW

**Review by kurtisvg — APPROVED** [review]
Great job.

File: samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+      // Each deny policy is attached to an organization, folder, or project.
+      // To work with deny policies, specify the attachment point.
+      //
+      // Its format can be one of the following:
+      // 1. cloudresourcemanager.googleapis.com/organizations/ORG_ID
+      // 2. cloudresourcemanager.googleapis.com/folders/FOLDER_ID
+      // 3. cloudresourcemanager.googleapis.com/projects/PROJECT_ID
+      //
+      // The attachment point is identified by its URL-encoded full resource name. Hence, replace
```
Comment (hegemonic) [inline samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java:64]: delete "full"

File: samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      // Construct the full path of the resource to which the policy is attached to.
+      // Its format is: "policies/{attachmentPoint}/denypolicies/{policyId}"
+      String policyParent = String.format("policies/%s/denypolicies/%s", attachmentPoint, policyId);
+
+      DenyRule denyRule =
+          DenyRule.newBuilder()
+              // Add one or more principals who should be denied the permissions specified in this
+              // rule.
+              // For more information on allowed values, see:
+              // https://cloud.google.com/iam/docs/principal-identifiers#v2
```
Comment (hegemonic) [inline samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java:79]: change to: https://cloud.google.com/iam/help/deny/principal-identifiers

File: samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+              .addDeniedPrincipals("principalSet://goog/public:all")
+
+              // Optionally, set the principals who should be exempted from the list of principals
+              // added in "DeniedPrincipals".
+              // Example, if you want to deny certain permissions to a group but exempt few
```
Comment (hegemonic) [inline samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java:84]: s/few/a few/

File: samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+              // principals, then add those here.
+              // .addExceptionPrincipals(
+              //     "principalSet://goog/group/project-admins@example.com")
+
+              // Set the permissions to deny.
+              // The permission value is of the format: service_fqdn/resource.action
+              // For the list of supported permissions, see:
+              // https://cloud.google.com/iam/docs/deny-permissions-support
```
Comment (hegemonic) [inline samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java:92]: change to: https://cloud.google.com/iam/help/deny/supported-permissions

File: samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                      // perform operations on projects that are tagged prod.
+                      // To solve this problem, you create a deny rule that denies the
+                      // cloudresourcemanager.googleapis.com/projects.delete permission to everyone
+                      // except project-admins@example.com for resources that are tagged prod.
+                      .setExpression("!resource.matchTag('12345678/env', 'prod')")
+                      .setTitle("Only for prod projects")
+                      .build())
+              .build();
+
+      // Set the policy resource path, version (etag) and the updated policy rules.
```
Comment (hegemonic) [inline samples/cloud-client/snippets/src/main/java/UpdateDenyPolicy.java:121]: s/policy rules/deny rules/

**Review by hegemonic — CHANGES_REQUESTED** [review]
Looks good overall! Most of these comments are very minor.

File: samples/cloud-client/snippets/src/main/java/CreateDenyPolicy.java [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+      // Its format can be one of the following:
+      // 1. cloudresourcemanager.googleapis.com/organizations/ORG_ID
+      // 2. cloudresourcemanager.googleapis.com/folders/FOLDER_ID
+      // 3. cloudresourcemanager.googleapis.com/projects/PROJECT_ID
+      //
+      // The attachment point is identified by its URL-encoded full resource name. Hence, replace
+      // the "/" with "%2F".
+      String attachmentPoint =
+          String.format("cloudresourcemanager.googleapis.com/projects/%s", projectId)
+              .replaceAll("/", "%2F");
```
[the remark this answers is not in the document]
Comment (Sita04) [reply samples/cloud-client/snippets/src/main/java/CreateDenyPolicy.java]: Modified it to use URLEncoder. Initially, thought of making the change obvious by just replacing the chars.

**Comment by Sita04** [comment]
@hegemonic PTAL.

File: samples/cloud-client/snippets/src/main/java/CreateDenyPolicy.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (hegemonic) [reply samples/cloud-client/snippets/src/main/java/CreateDenyPolicy.java]: Thanks! I understand your thinking, but I think this way is probably a little safer.

**Review by hegemonic — APPROVED** [review]
Looks great, Sita—thank you!

**Pull request merged**